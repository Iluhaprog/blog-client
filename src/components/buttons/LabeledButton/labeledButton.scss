@import '../../../vars.scss';

.labeled-button {
    font-size: 0.7em;
    background-color: $nav-bg;

    p {
        margin: 0;
        padding: 0;
    }

    p:first-child {
        position: relative;
        width: 20px;
        height: 20px;
        margin-right: 10px;

        span {
            position: absolute;
            width: 20px;
            height: 2px;
            background-color: $text-color-1;
            border-radius: 2px;
            top: 50%;
            left: 0;
            margin: 0;
            &:nth-child(2) {
                transform: rotate(90deg);
            }
        }
    }
}