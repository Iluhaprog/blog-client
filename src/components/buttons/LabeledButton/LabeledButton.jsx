import React from 'react';
import { Button } from '../Button';
import { Row } from '../../containers'
import './labeledButton.scss';

export default props => (
    <Button
        onClick={props.onClick}
        style='labeled-button'
    >
        <Row>
            <p>
                <span></span>
                <span></span>
            </p>
            <p>{props.text}</p>
        </Row>
    </Button>
)