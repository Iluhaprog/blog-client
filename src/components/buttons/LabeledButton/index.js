import LabeledButton from './LabeledButton';

export {
    LabeledButton,
};