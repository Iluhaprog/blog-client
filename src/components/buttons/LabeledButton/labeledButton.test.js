import React from 'react';
import { render } from '@testing-library/react';
import LabeledButton from './LabeledButton';

test('Test labeled button view', () => {
    const { asFragment } = render(<LabeledButton text='test'/>)
    expect(asFragment()).toMatchSnapshot();
});