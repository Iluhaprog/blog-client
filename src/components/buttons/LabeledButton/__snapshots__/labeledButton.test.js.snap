// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test labeled button view 1`] = `
<DocumentFragment>
  <button
    class="button default-button labeled-button"
  >
    <div
      class="row   "
    >
      <p>
        <span />
        <span />
      </p>
      <p>
        test
      </p>
    </div>
  </button>
</DocumentFragment>
`;
